File: cpp/Add-Strings/Add-Strings.cpp
/// Source : https://leetcode.com/problems/add-two-numbers/description/
/// Author : bryce
/// Time   : 2019-12-5

#include <iostream>
#include <string>
#include <math.h>
using namespace std;


class Solution {
public:
    string addStrings(string num1, string num2) {
        int i = num1.size()-1;
        int j = num2.size()-1;
        int carry = 0;
        string res;
        while (i>=0 || j>=0)
        {
            int temp = 0;
            if (i<0)
                temp = num2[j]-'0'+carry;
            if (j<0)
                temp = num1[i]-'0'+carry;
            else
                temp = num1[i] - '0' + num2[j] -'0' + carry;
            carry = temp / 10;
            res.insert(res.begin(),temp%10+'0');
            i--;
            j--;
        }
        if (carry>0)
            res.insert(res.begin(),carry+'0');
        return res;
    }
};

{
    string num1 = "9";
    string num2 = "9";
    string  res;
    res = Solution().multiply(num1,num2);
    cout << res << endl;
    system ("pause");
    return 0;
}

File: python/Add-Strings/Add-Strings.py
#! /usr/bin/env python
#! -*- coding=utf-8 -*-
# Date: 2019-12-5
# Author: Bryce

class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        res =""
        n1 = len(num1)
        n2 = len(num2)
        i = n1-1
        j = n2-1
        carry = 0
        while i >=0 or j >=0:
            if i < 0:
                temp = ord(num2[j]) - ord('0') + carry
            if j < 0:
                temp = ord(num1[i]) - ord('0') + carry
            if i>=0 and j>=0:
                temp = (ord(num1[i]) -ord('0')) + (ord(num2[j]) - ord('0')) + carry
            carry = temp // 10
            temp = temp%10
            res += str(temp)
            i=i-1
            j=j-1
        if carry>0:
            res+=str(carry)
        return res[::-1]
